spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -255,6 +260,28 @@ class SpdxCompoundExpression(
     override fun normalize(mapDeprecated: Boolean) =
         SpdxCompoundExpression(left.normalize(mapDeprecated), operator, right.normalize(mapDeprecated))
 
+    override fun sort(): SpdxExpression {
+        fun getSortedChildrenWithSameOperator(expression: SpdxCompoundExpression): List<SpdxExpression> {
+            val children = mutableListOf<SpdxExpression>()
+
+            fun addChildren(child: SpdxExpression) {
+                if (child is SpdxCompoundExpression && child.operator == operator) {
+                    children += getSortedChildrenWithSameOperator(child)
+                } else {
+                    children += child.sort()
+                }
+            }
+
+            addChildren(expression.left)
+            addChildren(expression.right)
+
+            return children
+        }
+
+        val children = getSortedChildrenWithSameOperator(this)
+        return parse(children.map { "($it)" }.sorted().joinToString(" ${operator.name} "))
```

Comment:
Can you maybe add some code comments about the idea of the algorithm?

Reply:
I was asking specifically as I thought `WITH` would still be an operator, but we changed that a while back. But what IÄm after: Would this break as soon as a non-commutative operator is added to `SpdxOperator`? Should we therefore explicitly check for `AND` and `OR`?

Reply:
I have rewritten the algorithm to use `reduce` instead of building and parsing a string for better efficiency. I think this should be self-explanatory now. Also it contains a `when` on the operator now which enforces revisiting the code if any new operators are added.

Reply:
I have also added a comment to the helper function now to explain what it does.